documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
+These modules will generate commands intended for the _Event Mesh_, which will then route these events to their designated endpoints.
+The _Event Mesh_ will allow for such transition, gradually making the system not only more responsive but also better suited to the real-world business logic.
+
+=== Problem Analysis
+
+Traditional systems often enforce strict transactional consistency, which can impede application performance and compromise business logic.
+For instance, upon the completion of a ride-sharing service at the end-user's destination, the system should reliably capture this real-world event.
+The capture should be performed regardless of any potential operational disruptions affecting dependent services (e.g., invoicing).
+
+In such scenarios, transactional applications typically return an error, which prevents any data from being changed, and causes the loss of real-world intent of end-users.
+This results in an adverse user experience and a deviation from the genuine business process, potentially leading to customer dissatisfaction.
+
+=== Solution Breakdown
+
+The concept of employing the _Event Mesh_ as a central, reliable hub for dispatching commands, as events, lies at the heart of this solution.
+This approach aligns closely with the Command Query Responsibility Segregation (_CQRS_) pattern, which distinctly categorizes commands and queries.
+Commands, in this context, are modeled as asynchronous events, designed to undergo processing by the _Event Mesh_.
+On the other hand, queries are synchronous operations, safe to retry, ensuring no loss of data integrity due to transient errors.
+
+The primary responsibility of the _Event Mesh_ is twofold.
+Firstly, it persists the incoming events, thereby maintaining a record of changes in the system's state.
+Secondly, it routes these events to their respective endpoints, ensuring that the appropriate microservices are notified and can subsequently update their internal states based on the event data.
+
+The mesh's inherent resilience is further bolstered by its exponential backoff strategy, which it employs when encountering operational failures.
+This mechanism ensures that the system retries the operation until it succeeds, thus mitigating the risk of data loss or system disruption due to transient issues.
+
+By integrating the _Event Mesh_ into the system architecture, several architectural benefits are achieved:
+
+* **Decomposition of the application into independently functioning services**:
+This approach facilitates a division of labor, with each service handling specific responsibilities.
+This not only enhances maintainability but also fosters scalability, as services can be independently scaled based on their demands.
 
-=== Using images 
+* **Improved business alignment**:
+By embracing an eventual consistency model, the _Event Mesh_ aligns closely with the inherent nature of most real-world business processes.
```

Comment:
This sounds very "teachy" :man_shrugging:

Reply:
The point of this was to show the eventual consistency fits better to the majority of business cases, instead of transactional approach. The transactional approach is widely miss-used, de-facto changing the business processes to force transactional behavior when it doesn't fit, for ex.: rollback of transaction when the invoicing application is unresponsive.

Do you think we can re-write to sound less "teachy" and still push the narrative?